src/renderers/common/Background.js

```diff
@@ -89,8 +85,11 @@ class Background extends DataMap {
 
 			if ( sceneData.backgroundCacheKey !== backgroundCacheKey ) {
 
-				sceneData.backgroundMeshNode.node = vec4( backgroundNode ).mul( backgroundIntensity );
+				const backgroundMeshNode = getBackgroundMeshNode( backgroundNode );
```

Comment:
@sunag Maybe there is a better way to fix this but using a fresh `backgroundMeshNode` the material is properly updated.

Reply:
Many of the current issues are related to https://github.com/mrdoob/three.js/pull/27904, the improvement brought better performance but broke some features that need to be reviewed.

Maybe just adding a `node.needsUpdate` could solve the problem?

```js
sceneData.backgroundMeshNode.needsUpdate = true;
```

```js
if ( sceneData.backgroundCacheKey !== backgroundCacheKey ) {

	sceneData.backgroundMeshNode.node = vec4( backgroundNode ).mul( backgroundIntensity );
	sceneData.backgroundMeshNode.needsUpdate = true;

	backgroundMesh.material.needsUpdate = true;

	sceneData.backgroundCacheKey = backgroundCacheKey;

}
```

Reply:
Indeed, that fixes it! I'll update the PR.